packages/supabase_flutter/lib/src/supabase_auth.dart

```diff
@@ -119,7 +119,12 @@ class SupabaseAuth with WidgetsBindingObserver {
       case AppLifecycleState.detached:
       case AppLifecycleState.inactive:
       case AppLifecycleState.paused:
-        Supabase.instance.client.auth.stopAutoRefresh();
+        // Realtime channels are kept alive in the background for some amount
+        // of time after the app is paused. If we stop refreshing the token
+        // here, the channels will be closed.
+        if (Supabase.instance.client.realtime.getChannels().isEmpty) {
+          Supabase.instance.client.auth.stopAutoRefresh();
+        }
```

Comment:
Hmm, the purpose of this piece of code is to lower the risks of creating an invalid JWT on the client side, and we should stop attempting to refresh the token when the app is in the background. 

If we could maybe stop subscribing to realtime through stream when the app is in background, that would be a more ideal implementation of this I would say.

Reply:
I think it's a bad idea to disconnect from realtime when going to background. On desktop for example everything keeps running, when going to background. On mobile you may often switch for a quick moment to another app. Disconnecting and reconnecting might miss events and causes overhead. I will do some further testing on this part.